src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -373,36 +375,30 @@ protected Blob fetchByPath(String tenantId, @Nonnull String path) {
         }
     }
 
-    protected String ensureRelativePath(String path) {
-        if (path.startsWith("/")) {
-            return path.substring(1);
-        }
-
-        return path;
-    }
-
     @Override
     public Optional<? extends Blob> findByPath(String path) {
         return findByPath(UserContext.getCurrentUser().getTenantId(), path);
     }
 
     @Override
     public Blob findOrCreateByPath(String tenantId, String path) {
-        if (Strings.isEmpty(path)) {
+        String sanitizedPath = utils.sanitizePath(path);
```

Comment:
I think this call has to be added to Blob::updateContent as well as we could otherwise overwrite a sanitized path with an unsanitized one by accident when updating the content of a blob and passing a file name

Reply:
updateContent only updates the file name. The only way to "unsanitize" the path is to pass an invalid file name, which is a separate problem?

Reply:
Yes thats exactly the problem, which when done will kill most parts of the storage system (vfs roots cant list any files of a space containing a single file with broken filename anymore)

We at least should create a ticket, my thought was that its at least related to the path sanitization (as the name seems to be created from the sanitized path in most cases, only the updateContent method somehow bypasses this logic.

Reply:
The ticket SIRI-251 should cover this

Reply:
Ah thx 👍